File: 3206_PhotoLesson_Backend/Phase 1/src/main/java/com/photolesson/backend/entity/Lecture.java
package com.photolesson.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "lectures")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Lecture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "section_id", nullable = false)
    private Section section;

    @Column(length = 255, nullable = false)
    private String title;

    @Column(name = "video_url", length = 500)
    private String videoUrl;

    @Column(name = "play_time")
    private Integer playTime;

    @Column(name = "sort_order")
    private Integer sortOrder;
}

File: 3206_PhotoLesson_iOS/Phase 1/3206_PhotoLesson/Models/Lecture.swift
import Foundation

struct Lecture: Codable, Identifiable {
    let lectureId: Int
    let title: String
    let videoUrl: String?
    let playTime: Int
    let sortOrder: Int?

    var id: Int { lectureId }

    var formattedPlayTime: String {
        let minutes = playTime / 60
        let seconds = playTime % 60
        return String(format: "%d:%02d", minutes, seconds)
    }
}

struct LectureDetail: Codable {
    let lectureId: Int
    let title: String
    let videoUrl: String?
    let playTime: Int
    let sectionId: Int?
}
